docs/content/sips/013-sqlite.md

```diff
@@ -0,0 +1,173 @@
+
+Like `wasi-sql` and the key-value store, we model resources such as database connections as pseudo-[resource handles](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#item-resource) which may be created using an `open` function and disposed using a `close` function. Each operation on a connection is a function which accepts a handle as its first parameter.
+
+Note that the syntax of the following `WIT` file matches the `wit-bindgen` version currently used by Spin, which is out-of-date with respect to the latest [`WIT` specification](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md) and implementation. Once we're able to update `wit-bindgen`, we'll update the syntax of all the Spin `WIT` files, including this one.
+
+```fsharp
+// A handle to an open sqlite instance
+type connection = u32
+
+// The set of errors which may be raised by functions in this interface
+variant error {
+  // A database with the supplied name does not exist
+  no-such-database,
+  // The requesting component does not have access to the specified database (which may or may not exist).
+  access-denied,
+  // The provided connection is not valid
+  invalid-connection,
+  // The database has reached its capacity
+  database-full,
+  // Some implementation-specific error has occurred (e.g. I/O)
+  io(string)
+}
+
+// Open a connection to a named database instance.
+//
+// If `database` is "default", the default instance is opened.
+//
+// `error::no-such-database` will be raised if the `name` is not recognized.
+open: func(name: string) -> expected<connection, error>
+
+// Execute a statement
+execute: func(conn: connection, statement: string, parameters: list<value>) -> expected<unit, error>
+
+// Query data
+query: func(conn: connection, query: string, parameters: list<value>) -> expected<query-result, error>
+
+// Close the specified `connection`.
+close: func(conn: connection)
+
+// A result of a query
+record query-result {
+  // The names of the columns retrieved in the query
+  columns: list<string>,
+  // The row results each containing the values for all the columns for a given row
+  rows: list<row-result>,
+}
+
+// A set of values for each of the columns in a query-result
+record row-result {
+  values: list<value>
+}
+
+// The values used in statements/queries and returned in query results
+variant value {
+  integer(s64),
+  real(float64),
+  text(string),
+  blob(list<u8>),
+  null
+}
+```
+
+*Note: the pseudo-resource design was inspired by the interface of similar functions in [WASI preview 2](https://github.com/bytecodealliance/preview2-prototyping/blob/d56b8977a2b700432d1f7f84656d542f1d8854b0/wit/wasi.wit#L772-L794).*
+
+#### Interface open questions
+
+**TODO**: answer these questions
+* `row-result` can be very large. Should we provide some paging mechanism or a different API that allows for reading subsets of the returned data?
```

Comment:
We could hedge on this by changing the function name to e.g. `query-all`, reserving `query` for a future cursor API.